test/e2e/hive.go

```diff
@@ -0,0 +1,74 @@
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/ARO-RP/pkg/api/admin"
+	"github.com/Azure/ARO-RP/pkg/hive"
+)
+
+var (
+	clusterPlatformLabelKey string = "hive.openshift.io/cluster-platform"
+	clusterRegionLabelKey   string = "hive.openshift.io/cluster-region"
+	clusterPlatformValue    string = "azure"
+
+	controlPlaneAPIURLOverride = func(clusterDomain string, clusterLocation string) string {
+		if !strings.ContainsRune(clusterDomain, '.') {
+			clusterDomain += "." + clusterLocation + "." + os.Getenv("PARENT_DOMAIN_NAME")
+		}
+
+		return fmt.Sprintf("api-int.%s:6443", clusterDomain)
+	}
+)
+
+var _ = Describe("Hive-managed ARO cluster", func() {
+	var oc *admin.OpenShiftCluster
+	var resourceID string
+
+	BeforeEach(func(ctx context.Context) {
+		resourceID = resourceIDFromEnv()
+		oc = adminGetCluster(Default, ctx, resourceID)
+
+		if oc.Properties.HiveProfile == (admin.HiveProfile{}) {
```

Comment:
We may want to make a function for this in `test/e2e/setup.go` like what we have for `skipIfNotInDevelopmentEnv()` and `skipIfDockerNotWorking()` that we can call inside of your test. This will be helpful if we add more tests that depend on the HiveProfile in the future.

Reply:
Made this change 👍🏻